forester/src/epoch_manager.rs

```diff
@@ -873,48 +877,55 @@ impl<R: RpcConnection, I: Indexer<R>> EpochManager<R, I> {
 
         let mut retries = 0;
         loop {
-            if !self
-                .check_eligibility(epoch_info, &work_item.tree_account)
-                .await?
-            {
-                debug!("Forester not eligible for this slot, skipping batch");
-                return Ok(None);
-            }
-
             match self
-                .process_transaction_batch(
-                    epoch_info,
-                    transaction_chunk,
-                    proof_chunk,
-                    indexer_chunk,
-                )
+                .check_eligibility(epoch_info, &work_item.tree_account)
                 .await
             {
-                Ok(signature) => {
-                    debug!(
-                        "Work item {:?} processed successfully. Signature: {:?}",
-                        work_item.queue_item_data.hash, signature
-                    );
-                    self.increment_processed_items_count(epoch_info.epoch.epoch)
-                        .await;
-                    return Ok(Some(signature));
+                Ok(_) => {
+                    match self
+                        .process_transaction_batch(
+                            epoch_info,
+                            transaction_chunk,
+                            proof_chunk,
+                            indexer_chunk,
+                        )
+                        .await
+                    {
+                        Ok(signature) => {
+                            debug!(
+                                "Work item {:?} processed successfully. Signature: {:?}",
+                                work_item.queue_item_data.hash, signature
+                            );
+                            self.increment_processed_items_count(epoch_info.epoch.epoch)
+                                .await;
+                            return Ok(Some(signature));
+                        }
+                        Err(e) => {
+                            if retries >= self.config.max_retries {
+                                error!(
+                                    "Max retries reached for work item {:?}. Error: {:?}",
+                                    work_item.queue_item_data.hash, e
+                                );
+                                return Err(e);
+                            }
+                            let delay = BASE_RETRY_DELAY * 2u32.pow(retries as u32);
```

Comment:
theoretic nit: could use saturating_pow to prevent overflows. anyway don't think it's important because retrys are not gonna overflow in practice

Reply:
Fixed.